tests/configlet/util/generic_patch.py

```diff
@@ -0,0 +1,217 @@
+    # Some keys could get updated by some control plane components
+    # Use this filter to mimic patch from NDM generated golden config.
+    #
+    filter = {
+            "ACL_TABLE",
+            "BGP_NEIGHBOR",
+            "BUFFER_PG",
+            "BUFFER_PORT_EGRESS_PROFILE_LIST",
+            "BUFFER_PORT_INGRESS_PROFILE_LIST",
+            "BUFFER_QUEUE",
+            "CABLE_LENGTH",
+            "DEVICE_NEIGHBOR",
+            "DEVICE_NEIGHBOR_METADATA",
+            "INTERFACE",
+            "PFC_WD",
+            "PORT",
+            "PORTCHANNEL",
+            "PORTCHANNEL_INTERFACE",
+            "PORTCHANNEL_MEMBER",
+            "PORT_QOS_MAP",
+            "QUEUE",
+            "VLAN_SUB_INTERFACE"
+    }
+    with open(os.path.join(src_dir, "config_db.json"), "r") as s:
+        src_json = json.load(s)
+
+    with open(os.path.join(dst_dir, "config_db.json"), "r") as s:
+        dst_json = json.load(s)
+
+    diff_patch = list(jsonpatch.JsonPatch.from_diff(src_json, dst_json))
+
+    jpatch = []
+
+    # Golden config will never add this. So required for test only
+    rm_pattern = "^/BUFFER_PG/Ethernet[0-9][0-9]*\|3-4"
+    for e in diff_patch:
+        if ((e["path"].split("/")[1] in filter) and
+                (not re.search(rm_pattern, e["path"]))):
+            jpatch.append(e)
+
+    if not hack_apply:
+        patch_file = os.path.join(patch_dir, "patch_0.json")
+        with open(patch_file, "w") as s:
+            s.write(json.dumps(jpatch, indent=4))
+    else:
+        # Hack:
+        # Split Interface off of tghe patch into seperate file
+        # Have this until generic patch updater will order 
+        # Interfaces add/remove correctly. Also remove port-admin-up
+        # as that should come at the end.
+        #
+        add_intf = False
+        interface_patch = []
+        other_patch = []
+
+        intf_pattern = ("^/INTERFACE/Ethernet[0-9][0-9]*$"
+                "|"
+                "^/INTERFACE/Ethernet[0-9][0-9]*\\|[0-9a-fA-F][0-9a-fA-F\\.:]*[0-9a-fA-F]~1[0-9][0-9]*$")
+
+        admin_up_pattern = "^/PORT/Ethernet[0-9][0-9]*/admin_status"
+
+        for e in jpatch:
+            if re.search(intf_pattern, e["path"]):
+                interface_patch.append(e)
+                add_intf = e["op"] == "add"
+            elif not re.search(admin_up_pattern, e["path"]):
+                other_patch.append(e)
+
+        patch_intf_file = os.path.join(patch_dir,
+                "patch_{}_intf.json".format(0 if add_intf else 1))
+        patch_other_file = os.path.join(patch_dir,
+                "patch_{}_other.json".format(1 if add_intf else 0))
+
+        with open(patch_intf_file, "w") as s:
+            s.write(json.dumps(interface_patch, indent=4))
+
+        with open(patch_other_file, "w") as s:
+            s.write(json.dumps(other_patch, indent=4))
+
+
+def _list_patch_files(patch_dir):
+    return sorted(fnmatch.filter(os.listdir(patch_dir), "patch_[0-9]_*.json"))
```

Comment:
I don't think this pattern will match the file name in the non-hack mode in line 92: `patch_0.json`

Reply:
good catch.